> This change seems to be causing warnings like the following during compilation with Visual Studio 2022 on the GitHub-runners: https://github.com/DrTimothyAldenDavis/SuiteSparse/actions/runs/9957325104/job/27509087596?pr=855#step:16:1022
> 
> ```
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(115): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(135): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(266): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(282): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(642): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(663): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(689): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(736): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(761): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(820): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(838): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(884): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(968): warning C4081: expected 'identifier'; found 'string'
> D:\a\SuiteSparse\SuiteSparse\UMFPACK\Source\umf_assemble.c(995): warning C4081: expected 'identifier'; found 'string'
> ```
> 
> That might mean that the pragma is ignored with that compiler.
> 
> Does it work correctly for you with Visual Studio 2017?

Sorry for the late reply.
I missed the MSVC warning above, and indeed I didn't encounter the previous compilation errors in Visual Studio 2017.

--- a/UMFPACK/Source/umf_internal.h
+++ b/UMFPACK/Source/umf_internal.h
@@ -170,7 +170,7 @@
 #  define UMFPACK_IVDEP _Pragma("ivdep")
 #elif defined (HAVE_PRAGMA_LOOP_IVDEP)
 #if defined ( _MSC_VER )
-#  define UMFPACK_IVDEP __pragma(loop( ivdep ))
+#  define UMFPACK_IVDEP __pragma(loop(ivdep))
 #else 
 #  define UMFPACK_IVDEP _Pragma("loop( ivdep )")
 #endif 
@@ -186,7 +186,7 @@
 #  define UMFPACK_NOVECTOR _Pragma("novector")
 #elif defined (HAVE_PRAGMA_LOOP_NO_VECTOR)
 #if defined ( _MSC_VER )
-#  define UMFPACK_NOVECTOR __pragma(loop( no_vector ))
+#  define UMFPACK_NOVECTOR __pragma(loop(no_vector))
 #else 
 #  define UMFPACK_NOVECTOR _Pragma("loop( no_vector )")
 #endif 

